tidy3d/components/base.py

```diff
@@ -731,7 +731,40 @@ def __eq__(self, other):
         """Define == for two Tidy3DBaseModels."""
         if other is None:
             return False
-        return self._json_string == other._json_string
+
+        def check_equal(dict1: dict, dict2: dict) -> bool:
+            """Check if two dictionaries are equal, with special handlings."""
+            if not dict1.keys() == dict2.keys():
+                return False
+
+            for key in dict1.keys():
+
+                val1 = dict1[key]
+                val2 = dict2[key]
+
+                if isinstance(val1, tuple) or isinstance(val2, tuple):
+                    val1 = dict(zip(range(len(val1)), val1))
+                    val2 = dict(zip(range(len(val2)), val2))
```

Comment:
`test_logging_warning_capture` seems to fail when it tries to compare two mediums with `frequency_range == None` and  `frequency_range != None`. We need to take into account cases when either `val1` or `val2` is `None`

Reply:
ok thanks for taking a look, I'd add that.

Reply:
adding 
```
if (val1 is None) ^ (val2 is None):
    return False
```
makes everything pass for me

Reply:
what if val1 is None and val2 is None, shouldn't it be `True`?

Reply:
oh wait. this is exclusive or I guess?

Reply:
yeah, apparently this is XOR in Python

Reply:
Note that this is a bit-wise XOR operator, not the logical one. Not that it matters in this case, but the logical XOR would be `(val1 is None) != (val2 is None)`